cpp/src/arrow/dataset/arrow-dataset.pc.in

```diff
@@ -18,7 +18,14 @@
 libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
+so_version=@ARROW_SO_VERSION@
+abi_version=@ARROW_SO_VERSION@
+full_so_version=@ARROW_FULL_SO_VERSION@
+
 Name: Apache Arrow Datasets
 Description: Apache Arrow Datasets provides an API to read and write semantic datasets stored in different locations and formats.
 Version: @ARROW_VERSION@
 Requires: arrow
+
```

Comment:
Could you remove this empty line? We don't have empty line here in other `.pc` such as `arrow-python.pc`.